src/Chapter1.hs

```diff
@@ -553,8 +552,11 @@ value after "=" where the condition is true.
 
 Casual reminder about adding top-level type signatures for all functions :)
 -}
-
-mid x y z = error "mid: not implemented!"
+mid :: Int -> Int -> Int -> Int
+mid x y z
+	| (x >= y && x <= z) || (x <= y && x >= z) = x
+	| (y >= x && y <= z) || (y <= x && y >= z) = y
+	| (z >= y && z <= x) || (z <= y && z >= x) = z
```

Comment:
As we mentioned, the compiler in Haskell is very attentive to the exhaustive pattern-matching. And here it would warn you that `Pattern matching is not exhaustive`, as the guards have quite complicated logic, and the compiler won't be able to prove that it covers all the cases.

Because of that, you will need to add another guard – `| otherwise = ...`, to tell the compiler, that your pattern matching is exhaustive 🙂 But actually, you can just replace this last line with 

```suggestion
	| otherwise = z
```
as this is the only possible answer in this case 👌🏼

Reply:
Oh, right! I forgot that. Thanks, I'll improve the solution!